[scs] Fix argument type of 'bwork', passed to QDLDL_factor.

https://github.com/cvxgrp/scs/issues/310

--- linsys/cpu/direct/private.c
+++ linsys/cpu/direct/private.c
@@ -51,7 +51,7 @@ static scs_int ldl_prepare(ScsLinSysWork *p) {
   L->i = (scs_int *)scs_calloc(nzmax, sizeof(scs_int));
   p->Dinv = (scs_float *)scs_calloc(n, sizeof(scs_float));
   p->D = (scs_float *)scs_calloc(n, sizeof(scs_float));
-  p->bwork = (scs_int *)scs_calloc(n, sizeof(scs_int));
+  p->bwork = (QDLDL_bool *)scs_calloc(n, sizeof(QDLDL_bool));
   p->fwork = (scs_float *)scs_calloc(n, sizeof(scs_float));
   return nzmax;
 }
--- linsys/cpu/direct/private.h
+++ linsys/cpu/direct/private.h
@@ -22,7 +22,8 @@ struct SCS_LIN_SYS_WORK {
   scs_int factorizations;
   /* ldl factorization workspace */
   scs_float *D, *fwork;
-  scs_int *etree, *iwork, *Lnz, *bwork;
+  scs_int *etree, *iwork, *Lnz;
+  QDLDL_bool *bwork;
   scs_float *diag_p;
 };

